src/backend/storeManagers/hyperplay/games.ts

```diff
@@ -1475,3 +1561,345 @@ function writeManifestFile(
+
+  logInfo(
+    `Patching ${gameInfo.title} from ${version} to ${newVersion}`,
+    LogPrefix.HyperPlay
+  )
+
+  try {
+    let totalBlocks = 0
+    let downloadedBlocks = 0
+    let downloadedData = 0
+    const blockSize = 512 * 1024 // 512KB in bytes
+    const startTime = Date.now()
+
+    const { generator } = patchFolder(
+      ipdtPatcher,
+      install_path,
+      currentManifest,
+      previousManifest,
+      ipfsGateway,
+      ipfsGateway,
+      signal,
+      maxWorkers ?? 6
+    )
+
+    if (signal.aborted) {
+      logWarning(`Patching ${appName} aborted`, LogPrefix.HyperPlay)
+      return { status: 'abort' }
+    }
+
+    signal.onabort = () => {
+      logWarning(`ON ABORT - Patching ${appName} aborted`, LogPrefix.HyperPlay)
+      aborted = true
+      return { status: 'abort' }
+    }
+
+    for await (const output of generator) {
+      logInfo(output, LogPrefix.HyperPlay)
+
+      if (signal.aborted) {
+        logWarning(`Patching ${appName} aborted`, LogPrefix.HyperPlay)
+        return { status: 'abort' }
+      }
+
+      const match = output.match(
+        /Blocks: (\d+)\/(\d+), Data Downloaded: ([\d.]+)/
+      )
+      if (match) {
+        downloadedBlocks = parseInt(match[1], 10)
+        totalBlocks = parseInt(match[2], 10)
+        downloadedData = parseInt(match[3])
+
+        const percent = (downloadedBlocks / totalBlocks) * 100
+        const currentTime = Date.now()
+        const elapsedTime = (currentTime - startTime) / 1000 // in seconds
+        const downloadedDataInMiB = downloadedData / 1024 / 1024
+        const downloadSpeed = downloadedData / elapsedTime
+        const totalSize = blockSize * totalBlocks
+        const eta = calculateEta(downloadedData, downloadSpeed, totalSize) ?? 0
+
+        sendFrontendMessage('gameStatusUpdate', {
+          appName,
+          status: 'patching',
+          runner: 'hyperplay',
+          folder: gameInfo.install.install_path
+        })
+
+        window?.webContents.send(`progressUpdate-${appName}`, {
+          appName,
+          status: 'patching',
+          runner: 'hyperplay',
+          folder: gameInfo.install.install_path,
+          progress: {
+            folder: gameInfo.install.install_path,
+            totalSize,
+            percent,
+            diskSpeed: downloadSpeed / 1024 / 1024,
+            downSpeed: downloadSpeed / 1024 / 1024,
+            bytes: downloadedDataInMiB,
+            eta
+          }
+        })
+
+        logInfo(
+          `Progress: ${percent.toFixed(2)}%, Downloaded: ${getFileSize(
+            downloadedData
+          )}, Speed: ${getFileSize(
+            downloadSpeed
+          )}/s, ETA: ${eta} totalSize: ${totalSize} = ${getFileSize(
+            totalSize
+          )}`,
+          LogPrefix.HyperPlay
+        )
+      }
+    }
+  } catch (error) {
+    logError(`Error while patching ${error}`, LogPrefix.HyperPlay)
+
+    trackEvent({
+      event: 'Patching Failed',
```

Comment:
do we already have some charts for this? I'd like to set up alerting

Reply:
No, just added these events for now. Didn't have time to create the charts yet but should be straightforward.